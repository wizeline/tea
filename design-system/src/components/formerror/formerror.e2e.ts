import { newE2EPage } from '@stencil/core/testing';

describe('wz-formerror', () => {
  it('renders form error', async () => {
    const page = await newE2EPage();
    await page.setContent("<wz-form-error error='Uh, oh'>Something</wz-input>");
    const errorText = await page.find('wz-error');
    expect(errorText.innerText).toEqual('Uh, oh');
  });
});
