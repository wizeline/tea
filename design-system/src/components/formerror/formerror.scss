wz-form-error {
  .error .input-container {
    border-color: var(--negative-base);
  }

  wz-error {
    display: block;
    height: 16px;
  }
}
