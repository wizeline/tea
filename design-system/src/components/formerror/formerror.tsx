import { Component, Prop } from '@stencil/core';

@Component({
  tag: 'wz-form-error',
  styleUrl: 'formerror.scss',
  shadow: false,
})
export class FormError {
  @Prop()
  error: string;

  render() {
    const errorClass = this.error ? 'error' : '';

    return (
      <div class={errorClass}>
        <slot />
        <wz-error>{Boolean(this.error) && this.error}</wz-error>
      </div>
    );
  }
}
