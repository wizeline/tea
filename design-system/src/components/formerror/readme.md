# wz-form-error



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description | Type     | Default     |
| -------- | --------- | ----------- | -------- | ----------- |
| `error`  | `error`   |             | `string` | `undefined` |


## Dependencies

### Used by

 - [wz-input](../input)
 - [wz-textarea](../textarea)

### Depends on

- [wz-error](../error)

### Graph
```mermaid
graph TD;
  wz-form-error --> wz-error
  wz-error --> wz-text
  wz-input --> wz-form-error
  wz-textarea --> wz-form-error
  style wz-form-error fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
