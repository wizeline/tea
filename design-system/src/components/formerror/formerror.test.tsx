import { TestWindow } from '@stencil/core/testing';
import { FormError } from './formerror';

describe('wz-input', () => {
  it('renders value', async () => {
    const window = new TestWindow();
    const element = await window.load({
      components: [FormError],
      html: "<wz-form-error error='Uh, oh'>Something</wz-input>",
    });
    const wzErrorText = element.querySelector('wz-error').textContent;

    expect(wzErrorText).toEqual('Uh, oh');
    expect(element).toMatchSnapshot();
  });
});
