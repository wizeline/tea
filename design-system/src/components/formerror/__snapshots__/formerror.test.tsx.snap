// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wz-input renders value 1`] = `
<wz-form-error
  class="hydrated"
  error="Uh, oh"
>
  
  
  <div
    class="error"
  >
    
    Something
    <wz-error>
      Uh, oh
    </wz-error>
  </div>
</wz-form-error>
`;
